lib/data/process-test-directory/index.js

```diff
@@ -30,6 +30,116 @@ const {
   toBuffer,
 } = require('./utils');
 
+// Regex Formats - Common regex patterns found in *.csvs
+const numericKeyFormat = /^_(\d+)$/;
+const idFormat = /^[A-Za-z0-3-:]+$/;
+const validReferenceKeys = /^(?:refId|type|value|linkText)$/;
+const validCommandKeys = /^(?:testId|command|settings|assertionExceptions|presentationNumber)$/;
+const assertionsExceptionsFormat = /^([0123]:[a-zA-Z-\d]+\s*)+$/;
+const settingsFormat = /^[A-Za-z0-9-\s]+$/;
+const validTestsKeys = /^(?:testId|title|presentationNumber|setupScript|instructions|assertions)$/;
+const titleFormat = /^[A-Z]([A-Za-z-',\s]){2,}[^.]$/;
+const assertionsFormat = /^([A-Za-z0-3-:]+\s*)+$/;
+
+const validAssertionsKeys = /^(?:assertionId|priority|assertionStatement|assertionPhrase|refIds)$/;
+const priorityFormat = /^[123]$/;
+// TODO: The make-v2 script isn't removing the punctuations at the end which violates the requirements of this
+//  column's validation. Based on https://github.com/w3c/aria-at/wiki/Test-Format-Definition-V2#assertionstatement
+//  Please ensure they are removing and use the commented regex instead.
+const assertionStatementFormat = /^[A-Z].*[a-zA-Z'{}().](?![.!?])$/;
+// const assertionStatementFormat = /^[A-Z].*[a-zA-Z'{}()](?![.!?])$/;
+
+// TODO: Same as TODO above. Based on https://github.com/w3c/aria-at/wiki/Test-Format-Definition-V2#assertionphrase
+const assertionPhraseFormat = /^[a-z].*[a-zA-Z'"{}()\d+.](?![.!?])$/;
+
+/**
+ * @param {object} row
+ * @param {string} row.testId
+ * @param {string} row.title
+ * @param {string} row.presentationNumber
+ * @param {string} [row.setupScript]
+ * @param {string} row.instructions
+ * @param {string} row.assertions
+ */
+function validateTestsKeys(row) {
+  for (const key of Object.keys(row)) {
+    if (numericKeyFormat.test(key)) {
+      throw new Error(`Column found without header row, ${+key.substring(1) + 1}`);
+    } else if (!validTestsKeys.test(key)) {
+      throw new Error(`Unknown tests.csv key: ${key} - check header row?`);
+    }
+  }
+  if (
+    !(
+      row.testId?.length &&
+      row.title?.length &&
+      row.presentationNumber?.length &&
+      row.instructions?.length &&
+      row.assertions?.length
+    )
+  ) {
+    throw new Error(
+      'Missing one of required testId, title, presentationNumber, instructions, assertions'
+    );
+  }
+  if (!idFormat.test(row.testId))
+    throw new Error('testId does not match the expected format: ' + row.testId);
+  if (!titleFormat.test(row.title))
+    throw new Error('title does not match the expected format: ' + row.title);
+  if (row.assertions && !assertionsFormat.test(row.assertions))
+    throw new Error('assertions does not match the expectedFormat: ' + row.assertions);
+  if (!Number(row.presentationNumber) > 0)
+    throw new Error(
+      'presentationNumber does not match the expected format: ' + row.presentationNumber
+    );
```

Comment:
@ChrisC replying to https://github.com/w3c/aria-at/pull/1247#discussion_r2139866711:

> There's an assertion ID named maximumValue255, so I think we don't actually want to limit numeric chars. @howard-e Let me know if you disagree?

Yep, we definitely don't want that since the regex covers the entire value, so agreed that changes in the last commit should be reverted.

I was more thinking of validating a unique case like that through the `validateTestsKeys` function similar to below:

```suggestion
    );
    
    if (row.assertions) {
      const assertionIds = row.assertions.split(' ');
      for (const assertionId of assertionIds) {
        const prefixMatch = assertionId.match(/^(\d+):/);
        if (prefixMatch) {
          const priority = parseInt(prefixMatch[1]);
          if (![0, 1, 2, 3].includes(priority)) {
            throw new Error(`assertions contains invalid priority "${priority}" for assertionId "${assertionId}". Only 0, 1, 2, or 3 are allowed: ` + row.assertions);
          }
        }
      }
    }
```

What do you think?

Reply:
This does seem little bit like over-engineering imo... besides, not restricting the numeric chars gives testers flexibility with the priority levels should they ever need it... but if you feel strongly about this, I can commit this and update the tests!

Reply:
@howard-e this should now be resolved with https://github.com/w3c/aria-at/pull/1247/commits/594d723026ea72bd78d5af095ac9bb845b430f14